src/Store/ExtensionClient.re

```diff
@@ -13,132 +13,164 @@ module LanguageFeatures = Feature_LanguageSupport.LanguageFeatures;
+          ~context=
+            Exthost.CompletionContext.{
+              triggerKind: Invoke,
+              triggerCharacter: None,
+            },
+          client,
+        )
+        |> Lwt.map(suggestionsToCompletionItems);
+      },
+    );
+};
 
-    let create =
-        (client, {id, selector}: Protocol.BasicProvider.t, (buffer, location)) => {
-      ProviderUtility.runIfSelectorPasses(
-        ~buffer,
-        ~selector,
-        () => {
-          let uri = Buffer.getUri(buffer);
-          let position = Protocol.OneBasedPosition.ofPosition(location);
-
-          ExtHostClient.provideDocumentHighlights(id, uri, position, client)
-          |> Lwt.map(definitionToModel);
-        },
-      );
-    };
+// TODO: Properly type definitoin provider...
+/*
+ module ExtensionDefinitionProvider = {
+   let definitionToModel = def => {
+     let Protocol.DefinitionLink.{uri, range, originSelectionRange} = def;
+     let Range.{start, _} = Protocol.OneBasedRange.toRange(range);
+
+     let originRange =
+       originSelectionRange |> Option.map(Protocol.OneBasedRange.toRange);
+
+     LanguageFeatures.DefinitionResult.create(
+       ~originRange,
+       ~uri,
+       ~location=start,
+     );
+   };
+
+   let create =
+       (client, {id, selector}: Protocol.BasicProvider.t, (buffer, location)) => {
+     ProviderUtility.runIfSelectorPasses(
+       ~buffer,
+       ~selector,
+       () => {
+         let uri = Buffer.getUri(buffer);
+         let position = Protocol.OneBasedPosition.ofPosition(location);
+         ExtHostClient.provideDefinition(id, uri, position, client)
+         |> Lwt.map(definitionToModel);
+       },
+     );
+   };
+ };
+ */
+
+module ExtensionDocumentHighlightProvider = {
+  let definitionToModel = (highlights: list(Exthost.DocumentHighlight.t)) => {
+    highlights
+    |> List.map(highlight =>
+         Exthost.OneBasedRange.toRange(
+           Exthost.DocumentHighlight.(highlight.range),
+         )
+       );
   };
 
-  module ExtensionFindAllReferencesProvider = {
-    let create =
-        (client, {id, selector}: Protocol.BasicProvider.t, (buffer, location)) => {
-      ProviderUtility.runIfSelectorPasses(
-        ~buffer,
-        ~selector,
-        () => {
-          let uri = Buffer.getUri(buffer);
-          let position = Protocol.OneBasedPosition.ofPosition(location);
-
-          ExtHostClient.provideReferences(id, uri, position, client);
-        },
-      );
-    };
+  let create =
+      (
+        id: int,
+        selector: Exthost.DocumentFilter.t,
+        client,
+        (buffer, location),
+      ) => {
+    ProviderUtility.runIfSelectorPasses(
+      ~buffer,
+      ~selector,
+      () => {
+        let uri = Buffer.getUri(buffer);
+        let position = Exthost.OneBasedPosition.ofPosition(location);
+
+        Exthost.Request.LanguageFeatures.provideDocumentHighlights(
+          ~handle=id,
+          ~resource=uri,
+          ~position,
+          client,
+        )
```

Comment:
Very minor, but since this patern is repeated may times in this module. These variables seem to just add unnecessary indirection:

```suggestion
        Exthost.Request.LanguageFeatures.provideDocumentHighlights(
          ~handle=id,
          ~resource=Buffer.getUri(buffer),
          ~position=Exthost.OneBasedPosition.ofPosition(location),
          client,
        )
```

Reply:
Went through and cleaned these up 👍